src/irmin/utils.ml

```diff
@@ -0,0 +1,17 @@
+let size_of t = Repr.unstage (Repr.size_of t)
+let to_bin_string t = Repr.(unstage (to_bin_string t))
+let pre_hash t = Repr.(unstage (pre_hash t))
+
+let old_pre_hash t =
```

Comment:
Since `old_pre_hash` is used only once, maybe you could move it back to `hash.ml`